### ISSUE TITLE
Mvc.JQuery.DataTables.Common's dependency on System.Linq.Dynamic.Core causes run-time error in an ASP.NET app

### ISSUE THREAD

**Issue by MrCsabaToth** [mcintyre321/mvc.jquery.datatables#165]
`Mvc.JQuery.DataTables.Common` (latest is 1.5.34) depends on `System.Linq.Dynamic.Core.1.0.6.13`. Problem is that `System.Linq.Dynamic.Core` now has version 1.0.7.5. Even though these are stepping version changes only, when I upgrade `System.Linq.Dynamic.Core` to the latest I get a run-time error:

> System.IO.FileLoadException: 'Could not load file or assembly 'System.Linq.Dynamic.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=0f07ec44de6ac832' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)'

When I try to create the return data:

     public DataTablesResult<CheckItemModel> GetCalculationCheckData(DataTablesParam dataTableParam)
        {
            ...
            var calculationChecks = EngineController.getCalculationChecks(id);
            var queryable = calculationChecks.AsQueryable().InterceptWith(new SetComparerExpressionVisitor(StringComparison.CurrentCultureIgnoreCase));
            return DataTablesResult.Create(queryable, dataTableParam, null);  // Here I get the error
        }

**Comment by MrCsabaToth** [comment]
I just checked and the PublicKeyToken of the .NET 4.6.1 assembly within the packages folder is indeed `0f07ec44de6ac832`.

**Comment by mcintyre321** [comment]
Binding redirects needed?

**Comment by MrCsabaToth** [comment]
@mcintyre321 Can you elaborate? After some search I have a feeling that I have to add an entry to the dependentAssemblies into the `web.config`?

**Comment by MrCsabaToth** [comment]
@mcintyre321 Can you elaborate? After some search I have a feeling that I have to add an entry to the dependentAssemblies into the `web.config`?

**Comment by MrCsabaToth** [comment]
I added

      <dependentAssembly>
        <assemblyIdentity name="System.Linq.Dynamic.Core" publicKeyToken="0f07ec44de6ac832" />
        <bindingRedirect oldVersion="0.0.0.0-1.0.7.5" newVersion="1.0.7.5" />
      </dependentAssembly>

but now I get an error during start of the app and not just later.

> Could not load file or assembly 'System.Linq.Dynamic.Core' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) 
>   Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 
>  Exception Details: System.IO.FileLoadException: Could not load file or assembly 'System.Linq.Dynamic.Core' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
> 
> Source Error: 
> 
>  An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.  
> 
> Assembly Load Trace: The following information can be helpful to determine why the assembly 'System.Linq.Dynamic.Core' could not be loaded.
> 
> === Pre-bind state information ===
> LOG: DisplayName = System.Linq.Dynamic.Core (Partial)
> WRN: Partial binding information was supplied for an assembly:
> WRN: Assembly Name: System.Linq.Dynamic.Core | Domain ID: 3
> WRN: A partial bind occurs when only part of the assembly display name is provided.
> WRN: This might result in the binder loading an incorrect assembly.
> WRN: It is recommended to provide a fully specified textual identity for the assembly,
> WRN: that consists of the simple name, version, culture, and public key token.
> WRN: See whitepaper http://go.microsoft.com/fwlink/?LinkId=109270 for more information and common solutions to this issue.
> LOG: Appbase = file:///C:/Users/Csaba/Documents/ProjSrc/Proj/
> LOG: Initial PrivatePath = C:\Users\Csaba\Documents\ProjSrc\Proj\bin
> Calling assembly : (Unknown).
> ===
> LOG: This bind starts in default load context.
> LOG: Using application configuration file: C:\Users\Csaba\Documents\ProjSrc\Proj\web.config
> LOG: Using host configuration file: C:\Users\Csaba\Documents\IISExpress\config\aspnet.config
> LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework\v4.0.30319\config\machine.config.
> LOG: Policy not being applied to reference at this time (private, custom, partial, or location-based assembly bind).
> LOG: Attempting download of new URL file:///C:/Users/Csaba/AppData/Local/Temp/2/Temporary ASP.NET Files/vs/29a2693a/701d5476/System.Linq.Dynamic.Core.DLL.
> LOG: Attempting download of new URL file:///C:/Users/Csaba/AppData/Local/Temp/2/Temporary ASP.NET Files/vs/29a2693a/701d5476/System.Linq.Dynamic.Core/System.Linq.Dynamic.Core.DLL.
> LOG: Attempting download of new URL file:///C:/Users/Csaba/Documents/ProjSrc/Proj/bin/System.Linq.Dynamic.Core.DLL.
> LOG: Using application configuration file: C:\Users\Csaba\Documents\ProjSrc\Proj\web.config
> LOG: Using host configuration file: C:\Users\Csaba\Documents\IISExpress\config\aspnet.config
> LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework\v4.0.30319\config\machine.config.
> LOG: Redirect found in application configuration file: 0.0.0.0 redirected to 1.0.7.5.
> LOG: Post-policy reference: System.Linq.Dynamic.Core, Version=1.0.7.5, Culture=neutral, PublicKeyToken=0f07ec44de6ac832
> LOG: Attempting download of new URL file:///C:/Users/Csaba/AppData/Local/Temp/2/Temporary ASP.NET Files/vs/29a2693a/701d5476/System.Linq.Dynamic.Core.DLL.
> LOG: Attempting download of new URL file:///C:/Users/Csaba/AppData/Local/Temp/2/Temporary ASP.NET Files/vs/29a2693a/701d5476/System.Linq.Dynamic.Core/System.Linq.Dynamic.Core.DLL.
> LOG: Attempting download of new URL file:///C:/Users/Csaba/Documents/ProjSrc/Proj/bin/System.Linq.Dynamic.Core.DLL.
> WRN: Comparing the assembly name resulted in the mismatch: Major Version
> ERR: Failed to complete setup of assembly (hr = 0x80131040). Probing terminated.

**Comment by MrCsabaToth** [comment]
If I modify the binding redirect (since it is trippin' about the major version) to

  <dependentAssembly>
    <assemblyIdentity name="System.Linq.Dynamic.Core" publicKeyToken="0f07ec44de6ac832" />
    <bindingRedirect oldVersion="0.0.0.0-1.0.7.5" newVersion="1.0.7.5" />
  </dependentAssembly>

then the app starts. But then I have the runtime error again. Maybe the Mvc.Jquery Datatables package needs a binding redirect internally as well?

![screenshot_2017-07-03_17-42-01](https://user-images.githubusercontent.com/744949/27811110-33fabd4a-6017-11e7-82d0-7088a39939a7.png)

**Comment by mcintyre321** [comment]
I think if this is going to cause lots of people problems, it's probably
worth updating the dependency withing Mvc.Jquery.DataTables - can you send
me a PR with the updated dependency? I'll merge it and get a new version
out ASAP.

**Comment by MrCsabaToth** [comment]
I made a PR, but I'm still thinking why it's needed. We are not supposed to update that every time there's a newer version. Take the `Newtonsoft.JSON` reference one line above the `System.Linq.Dynamic.Core`: Refers to 9.0.1, while the latest is now 10.x and my solution has the latest installed. That's not even a stepping version or minor version change but *major* version. Still the system is not trippin' about it. It cannot hurt though to increase the version, but I wonder if there's a problem with `System.Linq.Dynamic.Core`, or in general some problem as the older ASP.NET is slowly getting obsolete vs .NET.Core. I ran into trouble with `System.ValueTuples` as well with other packages, although those are just compile time warnings about ambiguity and don't cause any run-time error.
